llvm/lib/Target/AIE/AIE2TargetMachine.cpp

```diff
@@ -143,6 +143,7 @@ void AIE2PassConfig::addIRPasses() {
 }
 
 void AIE2PassConfig::addPreLegalizeMachineIR() {
+  addPass(createAIEAddressSpaceLowering());
```

Comment:
CHECK: we remove the extra information from the LLTs before doing anything else, which means that all our optimizations are based on the high level types as supplied in MMOs. As such this pass is 'repairing' the irtranslator to translate the annotated pointers to default LLT ptr types. We are pretty confident that we don't need more elaborate LLT types.
I can imagine we could keep distinct LLTs for PM, DM and TM, but that is not strictly necessary.

Reply:
PM, DM & TM has a distinct address space number, I would prefer keeping only P0 with different address space.